front-end/app/service/basket.service.ts

```diff
@@ -4,7 +4,7 @@ import { discountCalculator } from '../../../shared/discount-calculator'
 export class BasketService {
     items: BasketItemViewModel[] = [];
 
-    addToBasket(itemToAdd: ItemToAdd, selectedVersion?: string): void {
+    addToBasket(itemToAdd: ItemToAdd, selectedVersion?: string, tags?: string[]): void {
```

Comment:
Tags should always be on all items, even if it's just an empty array. Then you don't need to pass it in here as a 3rd parameter. You just have `tags: itemToAdd.tags`